sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -378,18 +515,50 @@ private static Context toAzureContext(ContextView context) {
      * @return The Reactor context.
      */
     public static reactor.util.context.Context toReactorContext(Context context) {
-        if (context == null) {
+        if (context == null || context == Context.NONE) {
             return reactor.util.context.Context.empty();
         }
+        return new AzureToReactorContextWrapper(context);
+    }
 
-        // Filter out null value entries as Reactor's context doesn't allow null values.
-        Map<Object, Object> contextValues = context.getValues().entrySet().stream()
-            .filter(kvp -> kvp.getValue() != null)
-            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+    private static final class AzureToReactorContextWrapper implements reactor.util.context.Context {
+        private Context azureContext;
+
+        AzureToReactorContextWrapper(final Context azureContext) {
+            this.azureContext = azureContext;
+        }
 
-        return CoreUtils.isNullOrEmpty(contextValues)
-            ? reactor.util.context.Context.empty()
-            : reactor.util.context.Context.of(contextValues);
+        @Override
+        @SuppressWarnings("unchecked")
+        public <T> T get(Object key) {
+            return (T) azureContext.getData(key).orElseThrow(() -> new NoSuchElementException());
+        }
+
+        @Override
+        public boolean hasKey(Object key) {
+            return azureContext.getData(key).isPresent();
+        }
+
+        @Override
+        public reactor.util.context.Context put(Object key, Object value) {
+            this.azureContext = azureContext.addData(key, value);
+            return this;
+        }
+
+        @Override
+        public reactor.util.context.Context delete(Object key) {
+            throw new UnsupportedOperationException("Deleting from this Reactor Context is not supported");
```

Comment:
If reactor context supports this, we should be supporting this too. Otherwise, this converted reactor context used elsewhere in the reactive stream which relies on this method will break.

Reply:
In Azure Context, we don't support delete. By transforming Azure context to Reactor context, we don't want to have delete anything, so I guess it will be fine if we leave as it is?

Reply:
I think this one and the transformation from ReactorContext to Azure Context both have this delete issue. Azure context is immutable (expected to be) and reactor context is not.